src/portfft/dispatcher/global_dispatcher.hpp

```diff
@@ -345,16 +345,16 @@ struct committed_descriptor_impl<Scalar, Domain>::run_kernel_struct<LayoutIn, La
         PORTFFT_LOG_TRACE("Dispatching the kernel for factor", factor_num, "of global implementation");
         if (static_cast<Idx>(factor_num) == dimension_data.num_factors - 1) {
           PORTFFT_LOG_TRACE("This is the last kernel");
-          l2_events =
-              detail::compute_level<Scalar, Domain, detail::layout::PACKED, detail::layout::PACKED, SubgroupSize>(
-                  current_kernel, desc.scratch_ptr_1.get(), desc.scratch_ptr_1.get(),
-                  desc.scratch_ptr_1.get() + imag_offset, desc.scratch_ptr_1.get() + imag_offset, twiddles_ptr,
-                  factors_and_scan, intermediate_twiddles_offset, impl_twiddle_offset, 0, committed_size,
-                  static_cast<Idx>(max_batches_in_l2), static_cast<IdxGlobal>(num_batches), static_cast<IdxGlobal>(i),
-                  static_cast<Idx>(factor_num), dimension_data.num_factors, storage, l2_events, desc.queue);
+          l2_events = detail::compute_level<Scalar, Domain, detail::layout::PACKED, detail::layout::PACKED,
+                                            SubgroupSize, const Scalar*>(
```

Comment:
Why do we have to give the final templater parameter explicitly here?

Reply:
Scratch pointers are non-const. We cast them to const here to match input, so that we have the same template instantiation of `compute_level` for scratch pointers and USM inputs (which are const). This way we can launch the same kernel for both cases.